src/adapters/implementations/paseto/token.service.ts

```diff
@@ -28,31 +28,47 @@ export class JWTAdapterService extends TokenAdapter {
 		super();
 	}
 
-	genAccess({
+	async genAccess({
 		accountId,
 		hasAcceptedLatestTerms,
-	}: GenAccessInput): GenAccessOutput {
+	}: GenAccessInput): Promise<GenAccessOutput> {
 		const payload: TokenPayload = {
 			sub: accountId,
 			terms: hasAcceptedLatestTerms,
 		};
 
+		const payloadRecord: Record<string, unknown> = {
+			sub: payload.sub,
+			terms: payload.terms,
+		};
+
 		const expiresAt = '';
 
-		const accessToken = this.jwt.sign(payload, this.config.get('JWT_SECRET'));
+		const secretString = await this.config.get('PASETO_SECRET');
+
+		const secretBuffer = Buffer.from(secretString, 'base64');
+		const secretKeyObject = this.paseto.bytesToKeyObject(secretBuffer);
+
+		const accessToken = await this.paseto.sign(payloadRecord, secretKeyObject);
 
 		return {
 			accessToken,
 			expiresAt,
 		};
 	}
 
-	validateAccess({ accessToken }: ValidateAccessInput): TokenPayload {
+	async validateAccess({
+		accessToken,
+	}: ValidateAccessInput): Promise<TokenPayload> {
 		try {
-			const payload = this.jwt.verify(
+			const secretString = this.config.get('PASETO_SECRET');
```

Comment:
Create a new "private" method (like `genAuthOutput` on `usecases/auth/auth.service.ts`) to get the paseto secret, since it`s used more than one time in the service.

Then, change the code to something like:
```
const secret = this.getSecret()
```